Source/WebKit/WebProcess/WebPage/WebPage.cpp

```diff
@@ -6620,6 +6621,14 @@ void WebPage::drawPagesForPrinting(FrameIdentifier frameID, const PrintInfo& pri
 
 void WebPage::addResourceRequest(WebCore::ResourceLoaderIdentifier identifier, const WebCore::ResourceRequest& request)
 {
+    ASSERT(!m_networkResourceRequestIdentifiersForPageLoadTiming.contains(identifier));
+    {
+        bool wasEmpty = m_networkResourceRequestIdentifiersForPageLoadTiming.isEmpty();
+        m_networkResourceRequestIdentifiersForPageLoadTiming.add(identifier);
+        if (wasEmpty)
+            send(Messages::WebPageProxy::StartNetworkRequestsForPageLoadTiming());
```

Comment:
Theoretically, we can send this IPC before adding the identifier to the map and avoid the `wasEmpty` local variable.

I'm also unclear why you wrapped all this new code in a new scope using curly brackets.

Reply:
Sure, will change the order.